### ISSUE TITLE
ingressClassName not supported, workaround?

### ISSUE THREAD

**Issue by lknite** [goharbor/harbor-helm#1085]
Since ingressClassName is not currently supported, I created my own ingress, and am unable to login with a basic helm install.  I expect this is true of most everyone with an up-to-date cluster attempting to install harbor.  This looks like what is needed for a 1.22.* cluster:  https://github.com/goharbor/harbor-helm/pull/1043

While we wait for ingressClassName to be supported, could someone provide the needed values to get this to work when someone has created their own ingress?  I came across a case which said I needed to specify 'externalUrl'.  I'm using a clusterIP (the result of the default helm install), so I've attempted:
```
externalUrl: http://harbor-core.harbor.svc.cluster.local
```
but that's not working ... logs from harbor-core don't show an attempt to connect when the login button is clicked

```
$ cat ingress/ingress.yaml 
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: harbor
  namespace: harbor
  annotations:
    #nginx.ingress.kubernetes.io/rewrite-target: /
    #nginx.ingress.kubernetes.io/proxy-body-size: 1000m
    ingress.kubernetes.io/ssl-redirect: 'true'
    ingress.kubernetes.io/proxy-body-size: '0'
    nginx.ingress.kubernetes.io/ssl-redirect: 'true'
    nginx.ingress.kubernetes.io/proxy-body-size: '0'
spec:
  ingressClassName: nginx
  tls:
  - hosts:
    - harbor.mydomain.net
    secretName: harbor
  rules:
  - host: harbor.mydomain.net
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: harbor-portal
            port:
              number: 80
```

```
$ cat values.yaml 
volumePermissions:
  enabled: true

externalURL: http://harbor-core.harbor.svc.cluster.local
#externalURL: http://harbor.mydomain.net

service:
  tls:
    enabled: true
    existingSecret: harbor
    commonName: harbor.mydomain.net

persistence:
  enabled: true
  #resourcePolicy:
```

**Comment by ninjadq** [comment]
did you try the #1043 pr solution? if you can verify it can resolve your issue, we will merge it

**Comment by sakomws** [comment]
Can we bump the issue please? We are blocked on upgrading our ingress? Looks the release was not cut from master yet, where ingressClassName is supported: https://github.com/goharbor/harbor-helm/blob/a57ee9e7672e03f3c605661109c5291c3819b511/templates/ingress/ingress.yaml#L59

**Comment by EE041B3-UnionInvestment** [comment]
please start the engines on 1.9.0 k8s 1.22 is not even the latest version anymore

**Comment by fredgate** [comment]
Branch `1.8` does not contain commit supporting ingress class name, 6 month ago.

**Comment by fredgate** [comment]
Can the commit https://github.com/goharbor/harbor-helm/commit/8fab6806f89fb385172f0cde318da1360ae9aa38 be included in release `1.8.2` ?

**Comment by aureq** [comment]
I think this has been shipped with 1.9.0 too according to https://github.com/goharbor/harbor-helm/blob/1.9.0/templates/ingress/ingress.yaml

Maybe this issue should be marked as closed.

**Comment by zyyw** [comment]
@aureq thanks for the info, I'm going to close this issue.

this is resolved in this PR:
* https://github.com/goharbor/harbor-helm/pull/1043
